fuzz/read_write_pkt.cc

```diff
@@ -464,37 +464,22 @@ ngtcp2_conn *setup_conn(FuzzedDataProvider &fuzzed_data_provider,
     }
   }
 
-  ngtcp2_transport_params remote_params{};
-
-  remote_params.initial_max_stream_data_bidi_local =
-    fuzzed_data_provider.ConsumeIntegralInRange<uint64_t>(0, NGTCP2_MAX_VARINT);
-  remote_params.initial_max_stream_data_bidi_remote =
-    fuzzed_data_provider.ConsumeIntegralInRange<uint64_t>(0, NGTCP2_MAX_VARINT);
-  remote_params.initial_max_stream_data_uni =
-    fuzzed_data_provider.ConsumeIntegralInRange<uint64_t>(0, NGTCP2_MAX_VARINT);
-  remote_params.initial_max_streams_bidi =
-    fuzzed_data_provider.ConsumeIntegralInRange<uint64_t>(0, NGTCP2_MAX_VARINT);
-  remote_params.initial_max_streams_uni =
-    fuzzed_data_provider.ConsumeIntegralInRange<uint64_t>(0, NGTCP2_MAX_VARINT);
-  remote_params.initial_max_data =
-    fuzzed_data_provider.ConsumeIntegralInRange<uint64_t>(0, NGTCP2_MAX_VARINT);
-  remote_params.active_connection_id_limit =
-    fuzzed_data_provider.ConsumeIntegralInRange<uint64_t>(
-      NGTCP2_DEFAULT_ACTIVE_CONNECTION_ID_LIMIT, NGTCP2_MAX_VARINT);
-  remote_params.max_udp_payload_size =
-    fuzzed_data_provider.ConsumeIntegralInRange<uint64_t>(
-      NGTCP2_MAX_UDP_PAYLOAD_SIZE, NGTCP2_MAX_VARINT);
-
-  rv = ngtcp2_transport_params_copy_new(&conn->remote.transport_params,
-                                        &remote_params, conn->mem);
+  auto chunk_len = fuzzed_data_provider.ConsumeIntegral<size_t>();
+  auto chunk = fuzzed_data_provider.ConsumeBytes<uint8_t>(chunk_len);
+
+  rv = ngtcp2_conn_decode_and_set_remote_transport_params(conn, chunk.data(),
+                                                          chunk.size());
   if (rv != 0) {
     ngtcp2_conn_del(conn);
     return nullptr;
   }
 
-  conn->local.bidi.max_streams = remote_params.initial_max_streams_bidi;
-  conn->local.uni.max_streams = remote_params.initial_max_streams_uni;
-  conn->tx.max_offset = remote_params.initial_max_data;
+  const ngtcp2_transport_params *remote_params =
+    ngtcp2_conn_get_remote_transport_params(conn);
+
+  conn->local.bidi.max_streams = remote_params->initial_max_streams_bidi;
+  conn->local.uni.max_streams = remote_params->initial_max_streams_uni;
+  conn->tx.max_offset = remote_params->initial_max_data;
```

Comment:
ngtcp2_conn_decode_and_set_remote_transport_params internally calls conn_sync_stream_id_limit which does L480-481. ngtcp2_conn_decode_and_set_remote_transport_params also does L482 indirectly.  We can omit L477-482.

Reply:
Fixed